### ISSUE TITLE
Fix S4056: Rule recommends overload with completely different signature 

### ISSUE THREAD

**Issue by adubadzel** [SonarSource/sonar-dotnet#2139]
### Description

The issue is raised on call of methods with several overloads when one of them has CultureInfo parameter, while others, with different signatures, don't. 

Is it expected behavior?
If yes, I would like to ask, if it can be configured somehow? And what use cases for that behavior are?
Now I have 3-party lib that has a class with such method and the method is used very often in my projects. So, every time I get false positive issue on it.

### Repro steps

```
string Get(int)
string Get(Enum, CultureInfo)
```

### Expected behavior

No error. 

### Actual behavior
False detection of Error S4056: Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.

### Related information

* SonarC# 7.9.1
* Scanner for MSBuild 4.0.2.892
